### PR TITLE
Add az_iot_get_status

### PR DESCRIPTION
**Pull request by danewalton**
While adding the twin parse API, I thought this seems like something we would want. We can set the enum value straight to the number from the response but that could be unsafe. Let me know your thoughts.

### CODE REVIEW

File: sdk/iot/core/inc/az_iot_core.h [hunk 1]
```diff
@@ -58,6 +58,19 @@ typedef enum
   AZ_IOT_STATUS_TIMEOUT = 504,
 } az_iot_status;
 
+/**
+ * @brief Get the #az_iot_status from an int
+ *
+ * @param[in] status_int  The int with the status number.
+ * @param[out]  status  The #az_iot_status* with the status enum.
+ * @return  The #az_result with the result of the get operation
```
Comment (ericwol-msft) [inline sdk/iot/core/inc/az_iot_core.h:66]: there are extra spaces here

Comment (danewalton) [reply sdk/iot/core/inc/az_iot_core.h:66]: I used tab instead of space. Will change.

File: sdk/iot/core/src/az_iot_core.c [hunk 2]
```diff
@@ -10,6 +10,71 @@
 
 #include <_az_cfg.h>
 
+az_result az_iot_get_status(uint32_t status_int, az_iot_status* status)
+{
+  switch(status_int)
```
Comment (ericwol-msft) [inline sdk/iot/core/src/az_iot_core.c:15]: nit, add a space

Comment (danewalton) [reply sdk/iot/core/src/az_iot_core.c:15]: running the clang formatter on the file

File: sdk/iot/core/src/az_iot_core.c [hunk 3]
```diff
@@ -10,6 +10,71 @@
 
 #include <_az_cfg.h>
 
+AZ_NODISCARD az_result az_iot_get_status(uint32_t status_int, az_iot_status* status)
```
Comment (ericwol-msft) [inline sdk/iot/core/src/az_iot_core.c:13]: ok, I have been trying to think of a better name for this...but I can't :(

Comment (danewalton) [reply sdk/iot/core/src/az_iot_core.c:13]: it could get `az_iot_get_status_from_int()`?

Comment (danewalton) [reply sdk/iot/core/src/az_iot_core.c:13]: or `..from_uint32()`. kinda like the span API's

Comment (danewalton) [reply sdk/iot/core/src/az_iot_core.c:13]: https://github.com/Azure/azure-sdk-for-c/blob/e789649ee5d56003078e52ac96d45710c6715fac/sdk/core/core/inc/az_span.h#L270

Comment (danewalton) [reply sdk/iot/core/src/az_iot_core.c:13]: updated

File: sdk/iot/core/inc/az_iot_core.h [hunk 4]
```diff
@@ -58,6 +58,19 @@ typedef enum
...
+ *
+ * @param[in] status_int The int with the status number.
+ * @param[out] status The #az_iot_status* with the status enum.
+ * @return The #az_result with the result of the get operation.
+ *  @retval #AZ_OK If the int is an #az_iot_status enum. `status` will be set to the according
+ * enum.
+ *  @retval #AZ_ERROR_ITEM_NOT_FOUND If the int is NOT an #az_iot_status enum.
+ */
+
+AZ_NODISCARD az_result az_iot_get_status_from_uint32(uint32_t status_int, az_iot_status* status);
```
Comment (CIPop) [inline sdk/iot/core/inc/az_iot_core.h:72]: I don't think this should be part of our public API.
The easiest way is to simply cast from an int/uint32_t. 

1. Why would we want the extra code on the device?
2. This is hiding the actual error if it's not in the list - I don't think that's a good idea.

**Pull request merged**